### PR TITLE
Document CWalletTx::mapValue entries and remove erase of nonexistent "version" entry.

### PR DESCRIPTION
**Pull request by ryanofsky**

### CODE REVIEW

**Comment by MarcoFalke** [comment]
Concept ACK, guess this could go to doxygen as well:
https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#doxygen-comments

**Comment by ryanofsky** [comment]
Thanks, switched to doxygen comment.

**Comment by sipa** [comment]
utACK

**Comment by fanquake** [comment]
utACK

**Review by jonasschnelli — APPROVED** [review]
Thanks for doing that!
ACK 28acf493f3c7cd461635a12e99fc3782dfe082a1

File: src/wallet/wallet.h [hunk 1]
```diff
@@ -248,6 +248,28 @@ class CWalletTx : public CMerkleTx
...
+    /**
+     * Key/value map with information about the transaction.
+     *
+     * The following keys can be read and written through the map and are
+     * serialized in the wallet database:
+     *
+     *     "comment", "to"   - comment strings provided to sendtoaddress,
+     *                         sendfrom, sendmany wallet RPCs
+     *     "from", "message" - obsolete fields that could be set in UI prior to
+     *                         2011 (removed 4d9b223)
```
Comment (paveljanik) [inline src/wallet/wallet.h]: 4d9b223 is the commit hash? github's UI can't find it. If it is commit hash, please add by/in before the ID.

File: src/wallet/wallet.h [hunk 2]
```diff
@@ -248,6 +248,28 @@ class CWalletTx : public CMerkleTx
...
+     *
+     * The following keys are serialized in the wallet database, but shouldn't
+     * be read or written through the map (they will be temporarily be added and
```
Comment (paveljanik) [inline src/wallet/wallet.h]: be... be...

File: src/wallet/wallet.h [hunk 3]
```diff
@@ -248,6 +248,28 @@ class CWalletTx : public CMerkleTx
...
+     * removed from the map during serialization):
+     *
+     *     "fromaccount"     - serialized strFromAccount value
+     *     "version"         - serialized nTimeSmart value
```
Comment (paveljanik) [inline src/wallet/wallet.h]: version.... nTimeSmart?

File: src/wallet/wallet.h [hunk 4]
```diff
@@ -248,6 +248,28 @@ class CWalletTx : public CMerkleTx
...
+     *     "n"               - serialized nOrderPos value
+     *     "timesmart"       - serialized nTimeSmart value
+     *     "spent"           - serialized vfSpent value that existed prior to
+     *                         2014 (removed 93a18a3)
```
Comment (paveljanik) [inline src/wallet/wallet.h]: removed by/in

File: src/wallet/wallet.h (hunk 1 above)
Comment (ryanofsky) [inline src/wallet/wallet.h]: Yeah not sure about the github interface. `git log 4d9b223` does work though.

File: src/wallet/wallet.h (hunk 2 above)
Comment (ryanofsky) [inline src/wallet/wallet.h]: Thanks, fixed.

File: src/wallet/wallet.h (hunk 3 above)
Comment (ryanofsky) [inline src/wallet/wallet.h]: Oops, fixed.

File: src/wallet/wallet.h (hunk 4 above)
Comment (ryanofsky) [inline src/wallet/wallet.h]: Thanks, added.

**Comment by paveljanik** [comment]
ACK 654e044

File: src/wallet/wallet.h [hunk 5]
```diff
@@ -248,6 +248,31 @@ class CWalletTx : public CMerkleTx
...
+     * be read or written through the map (they will be temporarily added and
+     * removed from the map during serialization):
+     *
+     *     "fromaccount"     - serialized strFromAccount value
+     *     "n"               - serialized nOrderPos value
+     *     "timesmart"       - serialized nTimeSmart value
+     *     "spent"           - serialized vfSpent value that existed prior to
+     *                         2014 (removed in commit 93a18a3)
+     *
+     * A mapValue.erase("version") statement also appears in the code,
```
Comment (laanwj) [inline src/wallet/wallet.h]: utACK - though would prefer to just remove this unnecessary code instead of bother to document it here

Comment (ryanofsky) [inline src/wallet/wallet.h]: Ok, added another commit to this PR which removes the mapValue.erase("version") statement and this part of the comment.

**Comment by ryanofsky** [comment]
New commit 87ed396159974b403d0ac00cb793888174e2665c adds documentation for bumpfee mapvalue entries (added in #8456)

**Pull request merged**